### PR TITLE
Feature suse 15.1

### PR DESCRIPTION
**Pull request by eradot4027**
#### Description:

- Continued adding support for the SUSE LInux Enterprise 15 platofrm.  Extended the standard profile, created a few oval checks for SUSE specific packages.  Updated other rules to support the platform.  In general mostly just checks that affect the SLE15 product.
#### Rationale:

- Adding support to the SSG for SLE15 for use in the absence of a certified DISA STIG.

### CODE REVIEW

File: shared/checks/oval/installed_OS_is_sle15.xml [hunk 1]
```diff
@@ -6,9 +6,9 @@
       <affected family="unix">
         <platform>multi_platform_all</platform>
       </affected>
-      <reference ref_id="cpe:/o:suse:linux_enterprise_server:15"
+      <reference ref_id="cpe:/o:suse:suse_linux_enterprise_server:15"
```
Comment (redhatrises) [inline shared/checks/oval/installed_OS_is_sle15.xml]: This CPE change is incorrect. Please keep the original.

File: shared/checks/oval/installed_OS_is_sle15.xml [hunk 2]
```diff
@@ -6,9 +6,9 @@
...
       source="CPE" />
-      <reference ref_id="cpe:/o:suse:linux_enterprise_desktop:15"
+      <reference ref_id="cpe:/o:suse:suse_linux_enterprise_desktop:15"
```
Comment (redhatrises) [inline shared/checks/oval/installed_OS_is_sle15.xml]: This CPE change is incorrect. Please keep the original.

File: sle15/cpe/sle15-cpe-dictionary.xml [hunk 3]
```diff
@@ -2,12 +2,12 @@
 <cpe-list xmlns="http://cpe.mitre.org/dictionary/2.0"
           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
           xsi:schemaLocation="http://cpe.mitre.org/dictionary/2.0 http://cpe.mitre.org/files/cpe-dictionary_2.1.xsd">
-      <cpe-item name="cpe:/o:suse:linux_enterprise_server:15">
+      <cpe-item name="cpe:/o:suse:suse_linux_enterprise_server:15">
```
Comment (redhatrises) [inline sle15/cpe/sle15-cpe-dictionary.xml]: This CPE change is incorrect. Please keep the original.

File: sle15/cpe/sle15-cpe-dictionary.xml [hunk 4]
```diff
@@ -2,12 +2,12 @@
...
             <title xml:lang="en-us">SUSE Linux Enterprise Server 15</title>
             <!-- the check references an OVAL file that contains an inventory definition -->
             <check system="http://oval.mitre.org/XMLSchema/oval-definitions-5" href="filename">installed_OS_is_sle15</check>
       </cpe-item>
-      <cpe-item name="cpe:/o:suse:linux_enterprise_desktop:15">
+      <cpe-item name="cpe:/o:suse:suse_linux_enterprise_desktop:15">
```
Comment (redhatrises) [inline sle15/cpe/sle15-cpe-dictionary.xml]: This CPE change is incorrect. Please keep the original.

File: ssg/constants.py [hunk 5]
```diff
@@ -262,7 +262,8 @@
         "cpe:/o:suse:linux_enterprise_server:12",
     ],
     "sle15": [
-        "cpe:/o:suse:linux_enterprise_server:15",
+        "cpe:/o:suse:suse_linux_enterprise_server:15",
+        "cpe:/o:suse:suse_linux_enterprise_desktop:15",
```
Comment (redhatrises) [inline ssg/constants.py]: This CPE change is incorrect. Please keep the original. The desktop cpe can be added, but it needs to be `cpe:/o:suse:linux_enterprise_desktop`

File: shared/checks/oval/installed_OS_is_sle15.xml (hunk 2 above)
Comment (eradot4027) [reply shared/checks/oval/installed_OS_is_sle15.xml]: Fixed

File: shared/checks/oval/installed_OS_is_sle15.xml (hunk 1 above)
Comment (eradot4027) [reply shared/checks/oval/installed_OS_is_sle15.xml]: Fixed

File: sle15/cpe/sle15-cpe-dictionary.xml [hunk 6]
```diff
@@ -67,9 +67,14 @@
...
       </cpe-item>
-      <cpe-item name="cpe:/a:yum">
-            <title xml:lang="en-us">Package yum is installed</title>
+      <cpe-item name="cpe:/a:zypper">
+            <title xml:lang="en-us">Package zypper is installed</title>
             <!-- the check references an OVAL file that contains an inventory definition -->
-            <check system="http://oval.mitre.org/XMLSchema/oval-definitions-5" href="filename">installed_env_has_yum_package</check>
+            <check system="http://oval.mitre.org/XMLSchema/oval-definitions-5" href="filename">installed_env_has_zypper_package</check>
+      </cpe-item>
+      <cpe-item name="cpe:/a:ssh:secure_shell">
```
Comment (redhatrises) [inline sle15/cpe/sle15-cpe-dictionary.xml]: This shouldn't be needed as the existing ssh checks should handle this.

File: shared/checks/oval/installed_env_has_openssh_package.xml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+<def-group>
+  <definition class="inventory"
+  id="installed_env_has_openssh_package" version="1">
```
[the remark this answers is not in the document]
Comment (eradot4027) [reply shared/checks/oval/installed_env_has_openssh_package.xml]: Concur, my bad.  Removed

File: sle15/cpe/sle15-cpe-dictionary.xml (hunk 6 above)
Comment (eradot4027) [reply sle15/cpe/sle15-cpe-dictionary.xml]: removed in latest commit

**Comment by redhatrises** [comment]
@openscap-ci add to whitelist

**Comment by redhatrises** [comment]
@openscap-ci test this please

**Pull request closed without merging**